components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterImplicits.scala

```diff
@@ -0,0 +1,15 @@
+package pl.touk.nussknacker.engine.api.definition
+
+object ParameterImplicits {
+
+  implicit class RichParameter(parameter: Parameter) {
+
+    def extract[T](params: Map[String, Any]): T =
```

Comment:
Do you know that we already have `ParameterWithExtractor` for that? Your implementation:
1. Is not connected with type used in `Parameter` creation
2. Doesn't use `ClassTag` so `asInstanceOf` won't throw an Exception in a fail-fast manner. Instead you will have suppressed exception during accessing some method/filed in the target class

I think that we should decide which one approach to use and to support only one of them.